packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+import { Provider } from '../zkEvm/types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+  passportProvider: Provider | null;
+  web3Provider: Web3Provider | null;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<string[]>;
+  loginCallback: () => Promise<void>;
+};
+
+const PassportContext = createContext<PassportContextType | null>(null);
+
+type PassportProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function PassportProvider({ children, config }: PassportProviderProps) {
```

Comment:
I'd recommend calling this something like ReactProvider to avoid confusion with the provider on line 17

Reply:
Considering the usage side, maybe `PassportReactProvider`?

Reply:
Normally you're going to be doing

```
import { passport } from '@imtbl/sdk';
```

so the double "passport" in

```
passport.PassportReactProvider
```

would be unnecessary IMO